packages/bot/src/struct/CommandHandler.ts

```diff
@@ -39,7 +41,13 @@ export class CommandHandler {
 		}
 
 		try {
-			const options = await command.handleAutocomplete(interaction);
+			const subcmd = interaction.options.getSubcommand(false);
+			const handleAutocomplete = subcmd
+				? this.commands.get(`${command.interactionOptions.name}-${subcmd}`)?.handleAutocomplete ??
+				  command.handleAutocomplete!
```

Comment:
1) These `command.handleAutocomplete` don't seem right even with prior checks in this function. It's possible we just have no `handleAutocomplete` on neither the subcommand nor the command.

2) The other mistake you're making is storing the method in a variable, which will make it lose its `this` context 👀

As such, this should probably be something like
```ts
const subcommandName = interaction.options.getSubcommand(false);
const subcommand = this.commands.get(`${command.interactionOptions.name}-${subcommandName}`);

const autocompleteHandler = subcommand?.handleAutocomplete ? subcommand : command;
const options = await autocompleteHandler.handleAutocomplete?.(interaction);
if (!options) {
	return await interaction.respond([]);
}

return await interaction.respond(options.slice(0, 25));
```